src/Command/ListCollections.php

```diff
@@ -79,8 +78,8 @@ public function __construct(string $databaseName, array $options = [])
             throw InvalidArgumentException::invalidType('"authorizedCollections" option', $options['authorizedCollections'], 'boolean');
         }
 
-        if (isset($options['filter']) && ! is_array($options['filter']) && ! is_object($options['filter'])) {
-            throw InvalidArgumentException::invalidType('"filter" option', $options['filter'], 'array or object');
+        if (isset($options['filter']) && ! is_document($options['filter'])) {
+            throw InvalidArgumentException::invalidType('"filter" option', $options['filter'], 'document');
```

Comment:
"document" is not a native php type. The term should be defined somewhere. Maybe update the message in `MongoDB\Exception\InvalidArgumentException`.

```diff
- Expected %s to have type "document" but found "string"
+ Expected %s to have type "document" (array or object) but found "string"
```

Reply:
Ah, yes. I was trying to avoid "array or object" since that would include all objects including `PackedArray`. I can add it if we want to glance over the packed array case. What do you think @jmikola?

Reply:
I thought I left a comment about my concern with using "document" as a type since we use actual PHP types everywhere else. I can't find that so perhaps I deleted the draft before submitting my last review.

In any event, I agree that something like `"document" (array or object)` would be preferable. We don't need to call out PackedArray specifically, as anyone using that should understand that it's not a document -- and if it leads to a bug report it'd be a nice opportunity to educate someone :)

Reply:
If you pass `document (array or object)` to the `invalidType()` factory method I suppose it's going to be wrapped in double quotes. Perhaps you just want to pass `document` and let the factory method deal with formatting -- but if that's a bit hairy (especially with the existing logic to handle arrays) then maybe just let it get quoted as-is.

Reply:
> Perhaps you just want to pass document and let the factory method deal with formatting

Reply:
I've created a new `expectedDocumentType` factory for this case - it seemed a more sensible choice than messing around with the formatting.